Clarify GlobalTransform::transform_point

masonk opened the pull request:
The existing doc comment for GlobalTransform::transform_point is unclear, or, arguably, incorrect. https://github.com/bevyengine/bevy/discussions/8501 also mentions this.

Additionally, a user reading the doc for transform_point might be looking for one of the three other transforms that I mentioned in this doc comment.

masonk commented:
Someone should check my logic :).

killercup commented on crates/bevy_transform/src/components/global_transform.rs:
```diff
@@ -189,9 +189,11 @@ impl GlobalTransform {
+    /// Transforms the given `point` from local space to global space, applying shear, scale, rotation and translation.
     ///
-    /// This moves `point` into the local space of this [`GlobalTransform`].
+    /// To transform a `point` from global to local, use `global_transform.affine().inverse().transform_point(point)`.
+    /// To transform a `vector` from local to global, use `global_transform.affine().transform_vector(vector)`.
+    /// To transform a `vector` global to local, use `global_transform.affine().inverse().transform_vector(vector)`.
```
I'd suggest using a code example (that is executed like a test so it must compile).

```suggestion
    /// It can be used like this:
    /// 
    /// ```
    /// # let global_transform = GlobalTransform::default();
    /// # let global_point = Vec3::ZERO;
    /// let local_point = global_transform.affine().inverse().transform_point3(global_point);
    /// # let local_vector = Vec3::ZERO;
    /// let global_vector = global_transform.affine().transform_vector3(local_vector);
    /// # let global_vector = Vec3::ZERO;
    /// let local_vector = global_transform.affine().inverse().transform_vector3(global_vector);
    /// ```
```

This also highlights that the methods are called `transform_vector3` and `transform_point3` :)

masonk replied:
Thanks for the suggestion. I wrote some doctests which are not the same ones that you wrote, please take a look.

masonk commented:
- I updated transform.rs that had similarly incorrect information
- Per [#](https://github.com/bevyengine/bevy/pull/14292#issuecomment-2227043407), made a pass on the struct comment
- Per https://github.com/bevyengine/bevy/pull/14292#discussion_r1676947108, I converted the comments into doctests.

killercup commented on crates/bevy_transform/src/components/global_transform.rs:
````diff
@@ -189,9 +188,39 @@ impl GlobalTransform {
+    /// assert_eq!(local_point, Vec3::new(1., 2., 3.))
+    /// ```
+    ///
+    /// To apply shear, scale, and rotation WITHOUT applying translation, different functions are available:
````
```suggestion
    /// To apply shear, scale, and rotation *without* applying translation, different functions are available:
```

maybe nicer typographically (but I don't really mind)

killercup reviewed (approved):
Great updates, thanks!
